toolchest_client/tools/tool.py

```diff
@@ -381,14 +387,17 @@ def run(self):
         temp_input_file_paths = []
         temp_output_file_paths = []
         non_parallel_output_path = f"{self.output_path}/{self.output_name}" if self.output_is_directory \
-            else self.output_path
+            and self.output_path else self.output_path
         for index, input_files in enumerate(jobs):
             # Add split files for merging and later deletion, if running in parallel
             temp_parallel_output_file_path = f"{self.output_path}_{index}"
             if should_run_in_parallel:
                 temp_input_file_paths += input_files
                 temp_output_file_paths.append(temp_parallel_output_file_path)
-            q = Query()
+
+            # Create a new Output for the thread.
+            self.output_locations[index] = Output()
```

Comment:
Nit: `output_object` and `output_locations` are hard to parse for someone – maybe even us – reading this code 6 months from now.

Is there a way you could rename this to be more specific (e.g. to provide meaning to what the `[index]` is, that the `output_location`s are tied to threads, etc)?

Reply:
Done, hopefully the refactored names are more clear now.

(Note: `Output.output_path` may be a bit unclear; this refers to the locally downloaded output path(s), if there are any. I didn't specify it as `local_output_path` because I think `output_path` on its own is straightforward enough for the user, but we can amend that if needed.)